### ISSUE TITLE
Incorrect cookie options passed to Express Session

### ISSUE THREAD

**Issue by iaincollins** [iaincollins/next-auth#12]
The following values should be set on the Express Session cookie option:

````
cookie: {
  httpOnly: true,
  secure: true
}
````

Currently it's set incorrectly:

````
  expressApp.use(expressSession({
    secret: sessionSecret,
    store: sessionStore,
    resave: false,
    rolling: true,
    saveUninitialized: false,
    httpOnly: true,
    cookie: {
      maxAge: sessionMaxAge
    }
  }))
````

The [documentation says that httpOnly is true by default](https://github.com/expressjs/session) in Express session, which is why it works as intended, even though the httpOnly option is specified in the wrong place.

I'd like to flip cookies to secure (HTTPS only) by default, though the secure option should still default to false for development and I need to figure out the best way to handle it. I might just expose it as an option for anyone that really wants to, but if not specified set it to 'true' if NODE_ENV=production (otherwise, leave it off).

**Comment by iaincollins** [comment]
Addendum: The documentation linked to above also notes that 'secure' can also be set to 'auto'.

**Comment by iaincollins** [comment]
This is now resolved, as of release 1.4.x, which now trusts proxy servers by default if they say the site is running behind SSL (e.g. on Heroku, or behind an nginx or haproxy server), which enables the new secure: 'auto' setting to work for cookies.

If a session cookie is set using HTTPS it will now ONLY work over HTTPS, but if if it's set in HTTP (e.g. when running locally in development) they will still work in HTTP.

Responsibility for enforcing HTTPS only (and automatic redirection from HTTP to HTTPS URLS) in production is left outside the scope of next-auth. As this is is how Heroku, now.sh, AWS (etc) and most private servers handle HTTP requests this seems like a reasonable approach that keeps things simple while defaulting to good behaviour.
